### PR TITLE
石川雄斗 EX8 解答 python

### PR DESCRIPTION
**Pull request by luna-moonlight1412**
# What

- VAEの学習と結果の可視化

実行コマンド
 `python main.py main`

| パラメータ    | デフォルト    | 説明         |
|-------------|-------------|-------------|
| --help      | ─           | ヘルプを表示します。 |
| --z_dim     | 2          | 潜在変数の次元 (2以上)を設定します。 |
| --h_dim     | 400          | 隠れ層の次元を設定します。 |
| --drop_rate     | 0.2          | ドロップアウト率を設定します。 |
| --learning_rate     | 0.001          | 学習率を設定します。 |
| --num_max_epochs     | 1000          | 最大エポック数を設定します。 |
| --do_train     | True          | 学習をするか否かを設定します。しない場合は`params/`の中のチェックポイントを使用します。 |
| --train_size_rate     | 0.8          | 訓練データとして使用するデータの割合 (0 ~ 1)を設定します。 |
| --seed     | 42          | ランダムシードを設定します。 |

# Result

z_2_0

z_2

z_2_0

# References [オプション]
[Variational Autoencoder徹底解説](https://qiita.com/kenmatsu4/items/b029d697e9995d93aa24)

### CODE REVIEW

**Review by MiyajiHikari — COMMENTED** [review]
@luna-moonlight1412 お疲れ様です。
基本的に、用意されていたコードの形式に沿った形できれいに穴埋めが出来ていると感じました！
ちなみに、課題内容の２で「このコードのままだと再現性がないという問題があるため，何を追加すればよいか考えてみましょう」とありましたが、これに対してどのようなアプローチを取ったか教えていただけますか？

**Comment by luna-moonlight1412** [comment]
@MiyajiHikari レビューありがとうございます。

すみません。課題2に気づいていませんでした。
今ランダムシードを`--seed`オプションで固定できるように変更しました。

**Review by MiyajiHikari — APPROVED** [review]
返信遅くなりすみません、確認しました。お疲れさまでした！
